### PR TITLE
Move natives to app runtime

### PR DESCRIPTION
**Pull request by jkoenig134**
# Readiness checklist

-   [x] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description

When migration from cordova to flutter we tried to make the natives less complex by splitting them down into "runtimeNatives". This was a way smaller effort then this - but we still couldn't get rid of the real complexity of the `@js-soft/native-abstraction` library. I've copied everything we really need from there to the app-runtime. This will allow us to archive the natives and have one repo less to maintain.

### CODE REVIEW

**Comment by jkoenig134** [comment]
Draft b/c it should be possible to remove many unused things.

File: packages/app-runtime/src/modules/AppRuntimeModule.ts [hunk 1]
```diff
@@ -12,14 +12,9 @@ export interface AppRuntimeModuleConfiguration extends ModuleConfiguration {}
...
-    protected subscribeToNativeEvent<TEvent>(event: Event, handler: EventHandler<TEvent>): void {
+    protected subscribeToNativeEvent<TEvent>(event: SubscriptionTarget<TEvent>, handler: EventHandler<TEvent>): void {
         const subscriptionResult = this.runtime.nativeEnvironment.eventBus.subscribe(event, handler);
-        if (subscriptionResult.isError) {
-            this.logger.error(subscriptionResult.error);
-            throw subscriptionResult.error;
-        }
-
-        this.nativeEventSubscriptionIds.push(subscriptionResult.value);
+        this.nativeEventSubscriptionIds.push(subscriptionResult);
```
Comment (Milena-Czierlinski) [inline packages/app-runtime/src/modules/AppRuntimeModule.ts:17]: What is the reason for these changes?

File: packages/app-runtime/src/natives/INativeDeviceInfoAccess.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Result } from "@js-soft/ts-utils";
+
+export enum PushServices {
+    "apns" = "apns", // eslint-disable-line @typescript-eslint/naming-convention
+    "fcm" = "fcm", // eslint-disable-line @typescript-eslint/naming-convention
+    "none" = "none" // eslint-disable-line @typescript-eslint/naming-convention
```
Comment (Milena-Czierlinski) [inline packages/app-runtime/src/natives/INativeDeviceInfoAccess.ts:6]: Can we add `"dummy"` here? (I remember we did some black magic a while ago to set that... https://github.com/nmshd/runtime/blob/ac65997fdaab9df917f46012e75ee6a57744a583/packages/app-runtime/test/mocks/NativeDeviceInfoAccessMock.ts#L14)

Comment (Milena-Czierlinski) [reply packages/app-runtime/src/natives/INativeDeviceInfoAccess.ts:6]: Wow, such a pretty formatting from git. 🤩

File: packages/app-runtime/src/natives/NativeEvents.ts
Comment (Milena-Czierlinski) [inline packages/app-runtime/src/natives/NativeEvents.ts:1]: I know the individual code snippets are small, but wouldn't is align better with our exiting code if we had a separate file for every event?

File: packages/app-runtime/src/natives/INativeDeviceInfoAccess.ts (hunk 2 above)
Comment (jkoenig134) [reply packages/app-runtime/src/natives/INativeDeviceInfoAccess.ts:6]: hmmmm, IMO we should do that later, but good catch!

File: packages/app-runtime/src/natives/NativeEvents.ts
Comment (jkoenig134) [reply packages/app-runtime/src/natives/NativeEvents.ts:1]: true!

File: packages/app-runtime/src/natives/INativeDeviceInfoAccess.ts (hunk 2 above)
Comment (jkoenig134) [reply packages/app-runtime/src/natives/INativeDeviceInfoAccess.ts]: no, NOW!

File: packages/app-runtime/src/modules/AppRuntimeModule.ts (hunk 1 above)
Comment (Milena-Czierlinski) [reply packages/app-runtime/src/modules/AppRuntimeModule.ts:17]: And this not only affects logging and throwing errors, but also lets you push a result and not its value?

Comment (jkoenig134) [reply packages/app-runtime/src/modules/AppRuntimeModule.ts:17]: I don't understand :D

Comment (jkoenig134) [reply packages/app-runtime/src/modules/AppRuntimeModule.ts:17]: it's only simplified

**Review by Milena-Czierlinski — APPROVED** [review]
